### DISCUSSION TITLE
Rolling Horizon Error

### DISCUSSION THREAD

**Discussion by hannesfelipe** [spine-tools/SpineOpt.jl#718] [Support]
Dear SpineOpt-Team, 
 
I recently installed SpineToolbox and SpineOpt. I am still getting into the environment, and I am happy to see all the options Spine provides. However, I’ve got a problem and don’t know how to solve it. 
 
I did all the steps from this tutorial: [https://www.youtube.com/watch?v=brf6rlzNDR0](https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3Dbrf6rlzNDR0&data=05%7C01%7Cjfg.eco%40cbs.dk%7C0f56a75dae464bd61b2d08db900a3dfa%7C875c414e5d004cdbb77adeae5d6ab201%7C0%7C0%7C638262145895669720%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=5zEjG3s0xOW0tXWvQAzhE3%2F1LRZ7aSsO8D3D2ovi%2Fpk%3D&reserved=0). Everything worked as described until the last element. Applying the roll_forward parameter leads to an error message. 
 
ERROR: LoadError: MethodError: no method matching roll!(::SpineInterface.TimeSlice, ::Dates.CompoundPeriod; update::Bool)
 
Closest candidates are:
roll!(::SpineInterface.TimeSlice, ::Union{Dates.CompoundPeriod, Dates.Period}; refresh) got unsupported keyword argument "update"
 
I assume that there has something changed in the Spine-structure since then? Or is it Julia specific. The tutorial uses Julia 1.8.5. I use Julia 1.9.2. I use spinetoolbox 0.6.18. 

Best regards and thanks for your support!

**Comment by Tasqu** [comment]
@hannesfelipe As far as I can tell, you seem to have a version conflict between SpineOpt and SpineInterface.
```julia
ERROR: LoadError: MethodError: no method matching roll!(::SpineInterface.TimeSlice, ::Dates.CompoundPeriod; update::Bool)

Closest candidates are:
roll!(::SpineInterface.TimeSlice, ::Union{Dates.CompoundPeriod, Dates.Period}; refresh) got unsupported keyword argument "update"
```
is caused by SpineOpt trying to call the `roll!` function using the `update` keyword, which was replaced by the `refresh` keyword by [this commit](https://github.com/spine-tools/SpineOpt.jl/commit/9cb2699f6efc995cf034ee258b78a37db7a0bd3c) on the 30th of June. Based on the given closest candidate using the `refresh` keyword, your SpineInterface seems to be up to date with SpineOpt lagging behind.

Updating your SpineOpt should hopefully fix this issue. Alternatively, you can try downgrading your SpineInterface to a version before [this commit](https://github.com/spine-tools/SpineInterface.jl/commit/0d030684e6789716f249b7493585cd8b45674063), but I wouldn't recommend that.

**Reply by hannesfelipe** [reply]
@Tasqu thank you for the respone. 

I already thought about this. My SpineOpt is version 0.06.13. Totally forgot to mention that. At least, I tried to install the current version of all packages. 

I assume that is the current version. Seems like I have to try the downgrading or do you have any other idea?

**Reply by Tasqu** [reply]
I'm guessing you've installed SpineOpt and SpineInterface according to the README instructions? E.g.
```julia
using Pkg
pkg"registry add General https://github.com/spine-tools/SpineJuliaRegistry"
pkg"add SpineOpt"
```
This can occasionally result in version conflicts between the two packages if we fail to properly update the `SpineJuliaRegistry`, which I'm guessing is the case here. I'll have to investigate.

In the meantime, you can try installing SpineOpt directly from the source *(instructions in the README)*, which will guarantee that you're using the latest development version of SpineOpt, which shouldn't have this issue with SpineInterface.
```julia
using Pkg
pkg"dev <path-to-your-local-SpineInterface-repository>"
pkg"dev <path-to-your-local-SpineOpt-repository>"
```
Note that you will have to download or clone the repositories on your machine for the above to work.

**Reply by Tasqu** [reply]
@hannesfelipe Sorry for the delay, I ran into another more serious issue with SpineInterface while figuring out this one. As I suspected, the issue was that the SpineOpt version 0.6.13 in the registry wasn't compatible with the SpineInterface version 0.11.3 in the registry. However, this should now be solved with versions 0.6.14 and 0.11.5 updated to the registry.

TLDR: `pkg> add SpineOpt` and `pkg> add SpineInterface` should now install compatible versions of SpineOpt and SpineInterface, avoiding this bug.

**Reply by hannesfelipe** [reply]
@Tasqu thank you very much for your support. I updated SpineOpt and SpineInterface today. Seems like everything is working now.
